### ISSUE TITLE
New Spring Element

### ISSUE THREAD

**Issue by epsi1on** [BriefFiniteElementNet/BriefFiniteElement.Net#167]
### Discussed in https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/discussions/166

<div type='discussions-op-text'>

<sup>Originally posted by **xyont** October 17, 2023</sup>
hi,

spring element is commonly used and basic of element, however it's not yet implemented or i'm missing? the availability of this element in the next or future version can be useful features. Many model such as beam or plate rested on elastic foundation can be solved.

regards,</div>

**Comment by epsi1on** [comment]
good idea, i think making a spring element like beam element, with so mproperty like `BarElement.Behaviour` which can be used to define axial, bending and torsional springs would be good.

**Comment by epsi1on** [comment]
there is an spring element named in `CustomElements` library named ParametricSpring ([link](https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/blob/master/Sources/CustomElements/Elements/ParametricSpring.cs)) which is good enough for usage. thus no need for new spring element

**Comment by MLiranzo** [comment]
Hello epsi11on,
Is it possible to use this class to support a plate or slab supported by an elastic floor?

**Comment by epsi1on** [comment]
> Hello epsi11on, Is it possible to use this class to support a plate or slab supported by an elastic floor?

Not sure, but I think so. Just note that small length EulerBernouly beams, behaves kind of different than expected. and I think Timoshenko beam should be used instead which I'm not sure be implemented yet. Same could be about triangle elements.

**Comment by MLiranzo** [comment]
Thank for the information.
